plugins/flytekit-flyin/flytekitplugins/flyin/jupyter_lib/decorator.py

```diff
@@ -0,0 +1,60 @@
+import subprocess
+import sys
+from functools import wraps
+from typing import Callable, Optional
+
+from flytekit.loggers import logger
+from .constants import NO_ACTIVITY_TIMEOUT
+
+
+def jupyter(
+    _task_function: Optional[Callable] = None,
+    no_activity_timeout: Optional[int] = NO_ACTIVITY_TIMEOUT,
+    token: Optional[str] = "",
+    port: Optional[int] = 8888,
+    enable: Optional[bool] = True,
+    notebook_dir: Optional[str] = "/root",
+    pre_execute: Optional[Callable] = None,
+    post_execute: Optional[Callable] = None,
+):
+    def wrapper(fn):
+        if not enable:
+            return fn
+
+        @wraps(fn)
+        def inner_wrapper(*args, **kwargs):
+            # 0. Executes the pre_execute function if provided.
+            if pre_execute is not None:
+                pre_execute()
+                logger.info("Pre execute function executed successfully!")
+
+            # 1. Launches and monitors the Jupyter Notebook server.
+            # Run the function in the background
+            logger.info("Start the jupyter notebook server...")
+            cmd = f"jupyter notebook --port {port} --NotebookApp.token={token}"
+            if notebook_dir:
```

Comment:
Do we need this line? notebook_dir has default value.

Reply:
updated